### PR TITLE
feat(devmanual): Document app maintainer role

### PR DESCRIPTION
**Pull request by ChristophWurst**
## Rendered

Bildschirmfoto vom 2023-04-17 19-15-54

### CODE REVIEW

File: developer_manual/app_publishing_maintenance/maintainer.rst
Comment (nickvergessen) [inline developer_manual/app_publishing_maintenance/maintainer.rst:33]: Talked recently with @AndyScherzinger about the "where to put" as being able to machine/script-read the information would be quite handy so we can automatically create reviews/alert etc in 3rd places, but for that we would need a dedicated format.

Two suggestions came to mind for that:

1. Extend the author information in info.xml to allow for a github handle: https://github.com/nextcloud/appstore/blob/master/nextcloudappstore/api/v1/release/info.xsd#L307-L314
2. Use the Codeowner file https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners#codeowners-file-location and require a line for `appinfo/info.xml`

This could then be used read e.g. when we create our security reviews and automatically mention the maintainers when their app is affected by something

**Comment by nickvergessen** [comment]
Otherwise 👍🏼

File: developer_manual/app_publishing_maintenance/maintainer.rst
Comment (ChristophWurst) [inline developer_manual/app_publishing_maintenance/maintainer.rst:25]: link to https://docs.nextcloud.com/server/latest/developer_manual/app_publishing_maintenance/app_upgrade_guide/index.html

File: developer_manual/app_publishing_maintenance/maintainer.rst
Comment (ChristophWurst) [reply developer_manual/app_publishing_maintenance/maintainer.rst:33]: I would prefer to leave this to the maintainers. If they want to set themselves as code owner in their assigned repos fine by me. We don't need to enforce that, do we?
Code owners will make most sense for apps inside the server repo

Comment (nickvergessen) [reply developer_manual/app_publishing_maintenance/maintainer.rst:33]:
```suggestion
To avoid the so called `bus factor <https://en.wikipedia.org/wiki/Bus_factor>`_, it is highly recommended to have more than one maintainer for each app. Apps can have one primary maintainer and one or more backup maintainers.

For transparency and to allow scripted reading of the information, it is recommended to declare the maintainers of an app in the app's repository ``.github/CODEOWNERS`` file with a specific line for ``/appinfo/info.xml``, so that others know whom to contact. See this `<notifications app pull request https://github.com/nextcloud/notifications/pull/1524/files>`_ for sample code.
```

Comment (AndyScherzinger) [reply developer_manual/app_publishing_maintenance/maintainer.rst:33]: Goal of the codeowner's file for apps would be to bind it to the info xml to the owners get notified / assigned on PRs that bump the version of the app _and_ to have it easily / standardized machine readable for the sec report generation

Comment (AndyScherzinger) [reply developer_manual/app_publishing_maintenance/maintainer.rst:33]: So please change from readme to codeowners

Comment (AndyScherzinger) [reply developer_manual/app_publishing_maintenance/maintainer.rst:33]: So please change from readme to codeowners

**Review by AndyScherzinger — CHANGES_REQUESTED** [review]

File: developer_manual/app_publishing_maintenance/maintainer.rst
Comment (ChristophWurst) [reply developer_manual/app_publishing_maintenance/maintainer.rst:33]: In my understanding we expect maintainers to see any PR, so they also see PRs that touch the version number. With the exception of new migrations, the version number will only be touched by the maintainers, no? Put differently: I'm okay if someone leverages CODEOWNERS if it eases their workflow, but I dislike making it a hard requirement.

The target of the developer documentation is any Nextcloud app in our open source community. Do we mandate these changes for all of the apps or only Nextcloud GmbH maintained ones? For the latter we can use an internal doc.

Comment (nickvergessen) [reply developer_manual/app_publishing_maintenance/maintainer.rst:33]: > Do we mandate these changes for all of the apps or only Nextcloud GmbH maintained ones? For the latter we can use an internal doc.

The latter for now, yes. Security team can only contact others via email anyway.

Comment (ChristophWurst) [reply developer_manual/app_publishing_maintenance/maintainer.rst:33]: Another machine-readable format for the maintainer and contact info could be the info.mail.author attribute: https://github.com/nextcloud/contacts/pull/3323/commits/afcd82e254cf533f3fa98c4dad73c42ccb035ac3. If someone sets a real email address it can be used to contact them for security topics. If it's a github noreply address the user can be pinged. Just an idea.

Comment (ChristophWurst) [reply developer_manual/app_publishing_maintenance/maintainer.rst:33]: CODEOWNERS has been put in place for Gmbh maintained apps. Is the rest of the generic maintainer role description ready to be merged?

**Review by AndyScherzinger — APPROVED** [review]
Else fine by me, since @ChristophWurst asked

**Pull request merged**
